game/ato/flightplans/shiprecoverytanker.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Iterator, Type
+from game.ato.flightplans.standard import StandardFlightPlan, StandardLayout
+from game.ato.flightplans.ibuilder import IBuilder
+from game.ato.flightplans.standard import StandardLayout
+from game.ato.flightplans.waypointbuilder import WaypointBuilder
+from game.ato.flightwaypoint import FlightWaypoint
+from game.utils import feet
+
+
+@dataclass(frozen=True)
+class RecoveryTankerLayout(StandardLayout):
+    nav_to: list[FlightWaypoint]
+    recovery_ship: FlightWaypoint
+    nav_from: list[FlightWaypoint]
+
+    def iter_waypoints(self) -> Iterator[FlightWaypoint]:
+        yield self.departure
+        yield from self.nav_to
+        yield self.recovery_ship
+        yield from self.nav_from
+        yield self.arrival
+        if self.divert is not None:
+            yield self.divert
+        yield self.bullseye
+
+
+class RecoveryTankerFlightPlan(StandardFlightPlan[RecoveryTankerLayout]):
+    @staticmethod
+    def builder_type() -> Type[Builder]:
+        return Builder
+
+    @property
+    def tot_waypoint(self) -> FlightWaypoint:
+        return self.layout.recovery_ship
+
+    @property
+    def mission_departure_time(self) -> timedelta:
+        return timedelta(hours=2)
```

Comment:
This is absolute, not relative to the TOT. A recovery tanker with a TOT of 1:50 will stay on station for 10 minutes. A tanker with a TOT of 2:00 will not do anything. I think you just want `return self.patrol_end_time` here.

Reply:
Thanks for catching this.  I had given the `recovery` task a `Last Waypoint` of the final carrier waypoint, and needed _something_ here for `TankerInfo` used later on (which unless I missed it, the final time isn't displayed anywhere?).  Do you feel like giving me the tanker a final waypoint was the right call?  If so, ideally we'd need to get the carrier's waypoints and speed down to here.

Reply:
> I think you just want `return self.patrol_end_time` here.

I've just went and looked at this, if I change it to `patrol_end_time` the relationship becomes recursive.  I'll change the two methods so it's more in line with the other flight plans.

Reply:
> Thanks for catching this. I had given the `recovery` task a `Last Waypoint` of the final carrier waypoint, and needed _something_ here for `TankerInfo` used later on (which unless I missed it, the final time isn't displayed anywhere?). Do you feel like giving me the tanker a final waypoint was the right call? If so, ideally we'd need to get the carrier's waypoints and speed down to here.

I discovered I was wrong here with regards to TankerInfo, and `None` is valid.  I can get rid of `patrol_end_time` all together I think.